extensions/interactions/GraphInput/directives/graph-viz.component.spec.ts

```diff
@@ -0,0 +1,1881 @@
+      },
+      {
+        x: 196.60939025878906,
+        y: 95.05902099609375,
+        label: ''
+      },
+      {
+        x: 20,
+        y: 20,
+        label: ''
+      }
+    ]);
+  });
+
+  it('should not add vertex when the user is not allowed to add a' +
+  ' vertex', () => {
+    component.state.currentMode = component._MODES.ADD_VERTEX;
+    component.state.mouseX = 20;
+    component.state.mouseY = 20;
+    component.interactionIsActive = true;
+    component.canAddVertex = false;
+
+    expect(component.graph.vertices).toEqual([
+      {
+        x: 150,
+        y: 50,
+        label: ''
+      },
+      {
+        x: 200,
+        y: 50,
+        label: ''
+      },
+      {
+        x: 150,
+        y: 100,
+        label: ''
+      },
+      {
+        x: 196.60939025878906,
+        y: 95.05902099609375,
+        label: ''
+      }
+    ]);
+
+    component.onClickGraphSVG();
+
+    // Function onClickGraphSVG() must only run if the interaction is active.
+    // Therefore the value of interactionIsActive is verified to be true before
+    // the rest of the lines are executed.
+    expect(component.interactionIsActive).toBe(true);
+    expect(component.graph.vertices).toEqual([
+      {
+        x: 150,
+        y: 50,
+        label: ''
+      },
+      {
+        x: 200,
+        y: 50,
+        label: ''
+      },
+      {
+        x: 150,
+        y: 100,
+        label: ''
+      },
+      {
+        x: 196.60939025878906,
+        y: 95.05902099609375,
+        label: ''
+      }
+    ]);
+  });
+
+  it('should not add vertex when interaction is not active', () => {
+    component.state.currentMode = component._MODES.ADD_VERTEX;
+    component.state.mouseX = 20;
+    component.state.mouseY = 20;
+    component.state.selectedVertex = 1;
+    component.state.selectedEdge = 1;
+    component.interactionIsActive = false;
+
+    expect(component.graph.vertices).toEqual([
+      {
+        x: 150,
+        y: 50,
+        label: ''
+      },
+      {
+        x: 200,
+        y: 50,
+        label: ''
+      },
+      {
+        x: 150,
+        y: 100,
+        label: ''
+      },
+      {
+        x: 196.60939025878906,
+        y: 95.05902099609375,
+        label: ''
+      }
+    ]);
+    expect(component.state.selectedVertex).toBe(1);
+    expect(component.state.selectedEdge).toBe(1);
+
+    component.onClickGraphSVG();
+
+    // Function onClickGraphSVG() must only run if the interaction is active.
+    // Therefore the value of interactionIsActive is verified to be false before
+    // testing if a vertice was not added.
+    expect(component.interactionIsActive).toBe(false);
+    expect(component.graph.vertices).toEqual([
+      {
+        x: 150,
+        y: 50,
+        label: ''
+      },
+      {
+        x: 200,
+        y: 50,
+        label: ''
+      },
+      {
+        x: 150,
+        y: 100,
+        label: ''
+      },
+      {
+        x: 196.60939025878906,
+        y: 95.05902099609375,
```

Comment:
Same comment as before -- please leave a comment if this is a random value or explanation if this is not random.